File: memory_test.cpp

#include <iostream>

int main() {
    int* numbers = new int[1000000];
    long long sum = 0;
    for (int i = 0; i < 1000000; ++i) {
        numbers[i] = i;
        sum += numbers[i];
    }

    std::cout << "Sum: " << sum << std::endl;

    delete[] numbers;
    return 0;
}

File: MemoryTest.java

import java.util.ArrayList;

public class MemoryTest {
    public static void main(String[] args) {
        ArrayList<Integer> numbers = new ArrayList<>();
        for (int i = 0; i < 1_000_000; i++) {
            numbers.add(i);
        }
        System.out.println("Sum: " + numbers.stream().mapToInt(Integer::intValue).sum());
        numbers = null;
        System.gc();
    }
}
